source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/Util.mm

```diff
@@ -13,3 +17,120 @@ void configVisibility(UIView *view, const std::shared_ptr<AdaptiveCards::BaseCar
+            imgView.translatesAutoresizingMaskIntoConstraints = NO;
+            [containerView addSubview:imgView];
+            [containerView sendSubviewToBack:imgView];
+            
+            if (img) {
+                // apply now if image is ready, otherwise wait until it is loaded (ACRView::observeValueForKeyPath)
+                applyBackgroundImageConstraints(backgroundImage, imgView, img);
+            }
+            
+            [imgView setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
+            [imgView setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisVertical];
+            [imgView setContentCompressionResistancePriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
+            [imgView setContentCompressionResistancePriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisVertical];
+        }
+    }
+}
+
+void applyBackgroundImageConstraints(const BackgroundImage *backgroundImageProperties, UIImageView *imageView, UIImage *image)
+{
+    UIView *superView = [imageView superview];
+    switch (backgroundImageProperties->GetMode()) {
+        case BackgroundImageMode::Repeat:
+        {
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:superView attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeCenterY relatedBy:NSLayoutRelationEqual toItem:superView attribute:NSLayoutAttributeCenterY multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:superView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:superView attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:superView attribute:NSLayoutAttributeLeading multiplier:1.0 constant:0].active = YES;
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationEqual toItem:superView attribute:NSLayoutAttributeTrailing multiplier:1.0 constant:0].active = YES;
+            
+            imageView.contentMode = UIViewContentModeScaleAspectFill;
+            break;
+        }
+        case BackgroundImageMode::RepeatHorizontally:
+        {
+            [NSLayoutConstraint constraintWithItem:imageView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:image.size.height].active = YES;
```

Comment:
does this work with image that is bigger than the container view?

Reply:
For Container/Column, it works just fine. The image is cropped and aligned properly. For the card, the card is actually expanded to hold the entire image. Not sure if this is proper behavior or if I should make it crop. For stretch, the image has always made the card bigger so I'm guessing this is probably ok.

---
In reply to: [263055631](https://github.com/Microsoft/AdaptiveCards/pull/2388#discussion_r263055631) [](ancestors = 263055631)